File: SEP3_T2/src/main/java/cinema/model/Layout.java
package cinema.model;

import java.util.HashMap;
import java.util.Map;

public class Layout {

    public char maxLetter;
    public int maxSeatInt;
    public int id;

    private static final Map<Integer,Layout> instance = new HashMap<Integer,Layout>();

    private Layout(char maxLetter, int maxSeatInt, int id) {
        this.id=id;
        this.maxSeatInt=maxSeatInt;
        this.maxLetter=maxLetter;
    }

    public static Layout create(int id, char maxLetter, int maxSeatInt) {
        return instance.computeIfAbsent(id,
                k -> new Layout(maxLetter, maxSeatInt, id)
        );
    }

    public static Layout getInstance(int id){

        return instance.get(id);
    }

    public char getMaxLetter() {
        return maxLetter;
    }

    public int getMaxSeatInt() {
        return maxSeatInt;
    }

    public void setMaxLetter(char maxLetter) {
        this.maxLetter = maxLetter;
    }

    public void setMaxSeatInt(int maxSeatInt) {
        this.maxSeatInt = maxSeatInt;
    }

}

File: SEP3_T1/Entities/Layout.cs
namespace Entities;

public class Layout
{
    public static readonly Dictionary<int, Layout> instance = new();


    private Layout(char maxLetter, int maxSeatInt, int id)
    {
        this.id = id;
        this.maxSeatInt = maxSeatInt;
        this.maxLetter = maxLetter;
    }

    public char maxLetter { get; set; }
    public int maxSeatInt { get; set; }
    public int id { get; set; }

    public static Layout Create(int id, char maxLetter, int maxSeatInt)
    {
        if (!instance.ContainsKey(id)) instance[id] = new Layout(maxLetter, maxSeatInt, id);

        return instance[id];
    }

    public static Layout GetInstance(int id)
    {
        instance.TryGetValue(id, out var layout);
        return layout;
    }
}